mindsdb/integrations/handlers/azure_blob_handler/azure_blob_handler.py

```diff
@@ -0,0 +1,364 @@
+        return self.handler.add_data_to_table(table_name, df)
+
+
+class AzureBlobHandler(APIHandler):
+    """
+    This handler handles connection and execution of the SQL statements on Azure Blob.
+    """
+
+    name = 'azureblob'
+    supported_file_formats = ['csv', 'tsv', 'json', 'parquet']
+
+    def __init__(self, name: str, **kwargs):
+        super().__init__(name)
+        """ constructor
+        Args:
+            name (str): the handler name
+        """
+
+        self.connection = None
+        self.is_connected = False
+        self._tables = {}
+        self.storage_account_name = None
+        self.account_access_key = None
+        self._files_table = ListFilesTable(self)
+        self.container_name = None
+
+        connection_data = kwargs.get('connection_data')
+        if 'storage_account_name' in connection_data:
+            self.storage_account_name = connection_data['storage_account_name']
+
+        if 'account_access_key' in connection_data:
+            self.account_access_key = connection_data['account_access_key']
+
+        if 'container_name' in connection_data:
+            self.container_name = connection_data['container_name']
+
+        if 'connection_string' in connection_data:
+            self.connection_string = connection_data['connection_string']
+
+    def connect(self) -> BlobServiceClient:
+        """ Set up any connections required by the handler
+        Should return output of check_connection() method after attempting
+        connection. Should switch self.is_connected.
+        Returns:
+            HandlerStatusResponse
+        """
+        if self.is_connected is True:
+            return self.connection
+
+        sas_token = generate_account_sas(
+            account_name=self.storage_account_name, 
+            account_key=self.account_access_key, 
+            resource_types=ResourceTypes(service=True, container=True, object=True), 
+            permission=AccountSasPermissions(read=True, list=True), 
+            expiry=datetime.now() + timedelta(hours=1)
+        )
+
+        blob_service_client = BlobServiceClient(
+            account_url=f"https://{self.storage_account_name}.blob.core.windows.net",
+            credential=sas_token
+        )
+        self.connection = blob_service_client
+        self.is_connected = True
+        return blob_service_client
+
+    def check_connection(self) -> StatusResponse:
+        """ Check connection to the handler
+        Returns:
+            HandlerStatusResponse
+        """
+        response = StatusResponse(False)
+
+        try:
+            client = self.connect()
+            client.get_account_information()
+            response.success = True
+
+        except Exception as e:
+            logger.error(f'Error connecting to Azure Blob: {e}!')
+            response.error_message = e
+
+        self.is_connected = response.success
+        return response
+
+    def native_query(self, query: str = None) -> Response:
```

Comment:
Are we missing the implementation here?

Reply:
yes.. I'll remove this function as well in next commit, almost missed this, thanks for pointing this Zoran!

Reply:
@ZoranPandovski Should we remove the `native_query()` method or add an implementation similar to S3?

Reply:
:eyes: